gh-138558: Improve description of `Interpolation.expression` in library docs (#139187)
python/cpython: Doc/library/string.templatelib.rst

   .. attribute:: expression
      :type: str

>       For interpolations created from t-string literals, :attr:`!expression`

Reviewer: On the fence about which sounds better out of 'from' or 'by'.
```suggestion
      For interpolations created by t-string literals, :attr:`!expression`
```

Author: Hrm... I think I like "by" better now that I'm thinking about it

>       is the source text found inside the curly braces, including any

Reviewer: ```suggestion
      is the source text found inside the curly brackets (``{`` & ``}``), including any
```

Author: Aren't they "curly braces", not "brackets"?

(Or is there some python doc standard here?)

Author: Update: haha, I checked, and it looks like there's no consistency even in the Python docs.

Here's a google search for ["curly braces" in Python docs](https://www.google.com/search?q=%22curly+braces%22+site%3Adocs.python.org&client=safari&sca_esv=c8c5cbf533aa5f5e&rls=en&ei=6zLQaI6PFqK50PEP-pj14AI&ved=0ahUKEwjOsN_nreqPAxWiHDQIHXpMHSwQ4dUDCBA&uact=5&oq=%22curly+braces%22+site%3Adocs.python.org&gs_lp=Egxnd3Mtd2l6LXNlcnAiIyJjdXJseSBicmFjZXMiIHNpdGU6ZG9jcy5weXRob24ub3JnSJsIUNABWNwHcAJ4AJABAJgBO6ABuQKqAQE2uAEDyAEA-AEBmAIAoAIAmAMAiAYBkgcAoAeOArIHALgHAMIHAMgHAA&sclient=gws-wiz-serp) vs same search for ["curly brackets"](https://www.google.com/search?q=%22curly+brackets%22+site%3Adocs.python.org&client=safari&sca_esv=c8c5cbf533aa5f5e&rls=en&ei=8jLQaObCLben0PEPpKODqAg&ved=0ahUKEwjmg6LrreqPAxW3EzQIHaTRAIUQ4dUDCBA&uact=5&oq=%22curly+brackets%22+site%3Adocs.python.org&gs_lp=Egxnd3Mtd2l6LXNlcnAiJSJjdXJseSBicmFja2V0cyIgc2l0ZTpkb2NzLnB5dGhvbi5vcmdI3NUEUNzOBFiC1QRwA3gAkAEAmAFEoAG2A6oBATi4AQPIAQD4AQGYAgCgAgCYAwCIBgGSBwCgB-gCsgcAuAcAwgcAyAcA&sclient=gws-wiz-serp)

Top Google hit for "curly braces" in the docs is inside [Format String Syntax](https://docs.python.org/3/library/string.html#format-string-syntax)

Top Google hit for "curly brackets" in the docs is [in the index itself](https://docs.python.org/3/genindex-Symbols.html)

Happy to go either way here! 🤷🏻‍♂️

Reviewer: @davepeck I'd use 'curly brackets', it's how we refer to them in the [index](https://docs.python.org/3/genindex-Symbols.html).

>       whitespace, but excluding the curly braces themselves. For manually

Reviewer: ```suggestion
      whitespace, but excluding the curly brackets themselves. For manually
```

>       created interpolations, :attr:`!expression` is an arbitrary string.

Reviewer: ```suggestion
      created interpolations, :attr:`!expression` is the arbitrary string provided when constructing the interpolation instance.
```

      It is recommended that developers manually creating
      :class:`!Interpolation` instances use valid Python expressions or leave

>       the string empty, but this is not enforced.

Reviewer: Use the active voice:

```suggestion
      We recommend using valid Python expressions or the empty string for the ``expression`` field of manually created :class:`!Interpolation` instances, although this is not enforced at runtime.
```

   .. attribute:: expression
      :type: str

      For interpolations created by t-string literals, :attr:`!expression`

>       is the source text found inside the curly braces (``{`` & ``}``),

Reviewer 2: It's not everything inside the curly braces. If there's an `=` sign (for a debug expression) or a format specifier or a conversion character, the expression ends before the corresponding character. Not sure if we should be mentioning that though. Probably yes?

Author: Ah good point. Will update.

Author: @lysnikolaou How do you like this update?

Reviewer 2: Looks good! Thanks @davepeck!
